sg14-error-codes.html

```diff
@@ -730,6 +784,17 @@
     some work in this area.
 </p>
 
+<!-- I'll be very blunt here, system_error is what it is, it can't be fixed now. The ship has sailed.
+
+Much more interesting is the development of a system_error2 living in the std2:: namespace
+
+If SG14 can champion a system_error2 fixing all the major problems with system_error, I think I would find this
+discussion much more productive. As it is, right now we are whinging about various problems in system_error
+which I am not sure are actually showstoppers per se, just rather unfortunate.
+
+A system_error2 proposal in the std2:: namespace I think would clarify the discussion enormously.
+It's easy to complain, must harder to come up with something truly better -->
```

Comment:
We're both trying to have it both ways.
You say the issues in this paper are "[maybe not] actually showstoppers per se" yet are looking forward to burning the whole thing down for std2.
I am looking backward to identify problems with the existing system_error, yet I actually agree that the ship has sailed and there's little realistic hope of getting anything changed.

All of these positions are useful. I don't think we'll ever see a "proposal fixing all the major problems with system_error" unless we can first draw up a list of what those "major problems" even *are*.  (And if the end result is that *there are no major problems* and all us blind men just didn't understand how to use the elephant in the first place, then excellent — less work for the Committee and we've all learned something!)

Reply:
I think there are two things going on here really.

The first is that system_error is defective in terms of correctness, and I think there is a reasonable chance that a defect fix against it can be successful even though some existing code might break. I'll post onto SG14 about this topic alone shortly.

The second is that if one did design a perfect system_error2, I think there is a reasonable chance that once complete, most people will compare the two and say "meh, not enough value add here". I think that system_error is *serviceable*, as in, it has its problems, but none of them are showstopper severe enough that a system_error2 is direly needed for the good of all C++. Hence, for me, rather than submit a document to WG21 whinging about problems, submit a document to WG21 with *solutions*. Much more positive.